### PR TITLE
feat: add support for git branch in templates

### PR DESCRIPTION
**Pull request by SuperQ**
Add a `.Branch` field to the text templates populated by the current git
branch.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

File: internal/pipe/git/git.go [hunk 1]
```diff
@@ -65,6 +66,10 @@ func getInfo(ctx *context.Context) (context.GitInfo, error) {
 }
 
 func getGitInfo() (context.GitInfo, error) {
+	branch, err := getBranch()
+	if err != nil {
+		return context.GitInfo{}, errors.Wrap(err, "couldn't get current branch")
```
Comment (caarlos0) [inline internal/pipe/git/git.go:71]:
```suggestion
		return context.GitInfo{}, fmt.Errorf("couldn't get current branch: %w", err)
```

Comment (SuperQ) [reply internal/pipe/git/git.go:71]: Thanks, I started building this on a branch before this new pattern was added.

File: internal/pipe/git/git.go [hunk 2]
```diff
@@ -123,6 +130,10 @@ func validate(ctx *context.Context) error {
 	return nil
 }
 
+func getBranch() (string, error) {
+	return git.Clean(git.Run("rev-parse", "--abbrev-ref", "HEAD", "--quiet"))
```
Comment (caarlos0) [inline internal/pipe/git/git.go:134]: I think there are no tests checking that this gets the correct output...

Comment (SuperQ) [reply internal/pipe/git/git.go:134]: This is the typical command I use to get the current branch. But you're right, I don't know how to mock a test for this. Some hints would be welcome. But I know it works. :smile: 

```
[git/goreleaser] $ git rev-parse --abbrev-ref HEAD --quiet
branch_template
```

Comment (caarlos0) [reply internal/pipe/git/git.go:134]: Yeah, it works, but as-is if we remove the method, or change it to something that doesn't work, no tests will break..

here's an example: https://github.com/goreleaser/goreleaser/blob/master/internal/pipe/git/git_test.go#L30-L42

Comment (SuperQ) [reply internal/pipe/git/git.go:134]: That's great, thanks.

**Review by caarlos0 — APPROVED** [review]

**Pull request merged**